mcs/tools/resgen/monoresgen.cs

```diff
@@ -371,17 +406,86 @@ class TxtResourceReader : IResourceReader {
 	public IDictionaryEnumerator GetEnumerator() {
 		return data.GetEnumerator ();
 	}
-	
+
+	public static string NextWord(ref string line) {
+		int i, j;
+		string keywd;
+		line = line.TrimStart ();
+		for (i = 0; i < line.Length && !Char.IsWhiteSpace (line [i]) && line [i] != ';'; )
+			i++;
+
+		if (i < line.Length) {
+			for (j = i; j < line.Length && Char.IsWhiteSpace (line [j]) && line [j] != ';'; )
+				j++;
+
+			keywd = line.Substring (0, i);
+			line = line.Substring (j).TrimStart ();
+		} else {
+			keywd = line;
+			line = "";
+		}
+		return keywd;
+	}
+
 	void Load () {
 		StreamReader reader = new StreamReader (s);
 		string line, key, val;
+		Stack<bool> conditional = new Stack<bool> (5);
 		int epos, line_num = 0;
+
+		conditional.Push(true);
```

Comment:
But in that case checking for `conditional.Peek () && stat` gets unnecessarily complicated.. my first implementation just pushed true or false and used `conditional.contains(false)`